libs/features/mas/src/hydrate.js

```diff
@@ -63,6 +63,15 @@ function processBadge(fields, merchCard) {
             'badge-background-color',
             fields.badgeBackgroundColor || DEFAULT_BADGE_BACKGROUND_COLOR,
         );
+        merchCard.setAttribute(
+            'border-color',
+            fields.badgeBackgroundColor || DEFAULT_BADGE_BACKGROUND_COLOR,
+        );
+    } else {
+        merchCard.setAttribute(
+            'border-color',
+            fields.borderColor || DEFAULT_BORDER_COLOR,
+        );
```

Comment:
i had logged https://jira.corp.adobe.com/browse/MWPW-167537 please have a look if you feel this could be taken now